uaclient/files/data_types.py

```diff
@@ -0,0 +1,68 @@
+import json
+from enum import Enum
+from typing import Any, Dict, Generic, Optional, Type, TypeVar
+
+import yaml
+
+from uaclient import exceptions
+from uaclient.data_types import DataObject
+from uaclient.files.files import UAFile
+
+
+class DataObjectFileFormat(Enum):
+    JSON = "json"
+    YAML = "yaml"
+
+
+DOFType = TypeVar("DOFType", bound=DataObject)
+
+
+class DataObjectFile(Generic[DOFType]):
+    def __init__(
+        self,
+        data_object_cls: Type[DOFType],
+        ua_file: UAFile,
+        file_format: DataObjectFileFormat = DataObjectFileFormat.JSON,
+    ):
+        self.data_object_cls = data_object_cls
+        self.ua_file = ua_file
+        self.file_format = file_format
+
+    def preprocess_data(self, data: Dict[str, Any]) -> Dict[str, Any]:
+        return data
```

Comment:
what do you think of making `preprocess_data` an argument instead of a method?
```suggestion
    def __init__(
        self,
        data_object_cls: Type[DOFType],
        ua_file: UAFile,
        file_format: DataObjectFileFormat = DataObjectFileFormat.JSON,
        preprocess_data: Optional[Callable[[Dict], Dict]] = None,
    ):
        self.data_object_cls = data_object_cls
        self.ua_file = ua_file
        self.file_format = file_format
        self.preprocess_data = preprocess_data
```

Reply:
Good point, I will make the change